### PR TITLE
Add flatmap service, and refine flatmap imagegallery results

### PR DESCRIPTION
**Pull request by Tehsurfer**
# Description

This is for a sprint 19 QA ticket:
[Clicking on a flatmap from within a dataset should take user to relevant anatomical structure on said flatmap](https://www.wrike.com/open.htm?id=859089097)

https://jesse-sprint-19.herokuapp.com/datasets/125?type=dataset&datasetDetailsTab=images
image

Changes: 

 #### Added a `flatmaps` service for querying the flatmap knowledgbase. 
 This is required to check if the anatomy exists for a given species on a flatmap. (A request of the ticket was to not open the flatmap if the anatomy was not annotated)
 
 #### Added thumbnails that correspond to the species of the dataset
 Previously the thumbnails were always the rat flatmap. Now there is a thumnail for each species 

## Type of change

Delete those that don't apply.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

Tested locally and can be tested at 
https://jesse-sprint-19.herokuapp.com/datasets/125?type=dataset&datasetDetailsTab=images

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have utilized components from the Design System Library where applicable
- [ ] I have added unit tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: pages/datasets/_datasetId.vue [hunk 1]
```diff
@@ -294,24 +295,43 @@ const getThumbnailData = async (datasetDoi, datasetId, datasetVersion, datasetFa
...
-            id: datasetId,
-            version: datasetVersion
+
+        // Check if flatmap has the anatomy for this species. This is done by asking the flatmap knowledge base
+        // if a flatmap of (species) has (anatomy)
+        let foundAnatomy = []
+        if (scicrunchData.organs[0]) { // Check if dataset has organ annotation
+          const anatomy = scicrunchData.organs.map(organ => organ.curie)
+          let data = await flatmaps.anatomyQuery(taxo, anatomy)
+          let anatomyResponse = data.data.values
```
Comment (alan-wu) [inline pages/datasets/_datasetId.vue:308]: Does the response contain data even in failure? Better check or the page would not have the SciCrunch and Biolucida data...

File: pages/datasets/_datasetId.vue [hunk 2]
```diff
@@ -294,24 +295,43 @@ const getThumbnailData = async (datasetDoi, datasetId, datasetVersion, datasetFa
...
+          if (anatomyResponse && anatomyResponse.length > 0) {
+            foundAnatomy = anatomyResponse.map(val => val[1]) // uberon is stored in second element of tuple
           }
-          flatmapData.push(organData)
+        }
 
-        });
+        // Add flatmaps that match the anatomy and taxonomy to the gallery
+        scicrunchData.organs.forEach(organ => {
+          if (foundAnatomy.includes(organ.curie)){
```
Comment (alan-wu) [inline pages/datasets/_datasetId.vue:316]: Does the foundAnatomy contain other curies beside the one provided in the request?

**Review by alan-wu — CHANGES_REQUESTED** [review]
Looks good but required to revert one local change and I have two questions about the changes. Thanks,

File: pages/datasets/_datasetId.vue (hunk 2 above)
Comment (Tehsurfer) [reply pages/datasets/_datasetId.vue:316]: No, it only contains the ones found in the request. 

It is just there for readability and to handle the case where the flatmap query returns `[[]]`

File: pages/datasets/_datasetId.vue (hunk 1 above)
Comment (Tehsurfer) [reply pages/datasets/_datasetId.vue:308]: yeah good question I will have a look at this

**Review by alan-wu — APPROVED** [review]

**Pull request merged**